src/common/transformations/src/transformations/op_conversions/convert_sequences_to_tensor_iterator.cpp

```diff
@@ -54,7 +60,6 @@ ngraph::Output<ngraph::Node> get_masked_value(const std::shared_ptr<ngraph::opse
     body_results.push_back(aggregated_result);
 
     auto scalar_mask_value = ngraph::opset5::Constant::create(data.get_element_type(), {}, {0.f});
-    auto data_shape = ngraph::op::util::make_try_fold<ngraph::opset5::ShapeOf>(data);
```

Comment:
Can you add new tests to cover changes?

Reply:
we already have functional tests for this transformation, I hope it's enough if they pass
the issue in the ticket is reproducible only via MO pipeline

Reply:
But what is the difference between runtime and MO pipeline?

Reply:
@ilyachur I meant reading the already converted IR via IE and MO.
if we load IR via MO IR Reader, we use MO shape inference functions for operations, it looks like they are slightly different from shape inference in IE. IE works well and without this fix. Maybe we also need to take a look at MO to check the difference, but this fix is also needed because "legacy_broadcast_for_binary_operation" is legacy and this is the only place where it's actually used.
I will delete this function in the next commit.